File: 151001/Novik/DC/src/Publisher/Mapper/NewsMapper.cs
using AutoMapper;
using Publisher.Models.DTOs.Requests;
using Publisher.Models.DTOs.Responses;
using Publisher.Models.Entity;

namespace Publisher.Mapper;

public class NewsMapper : Profile
{
    public NewsMapper()
    {
        CreateMap<News, NewsResponseTo>()
            .ForMember(dest => dest.id, opt => opt.MapFrom(src => src.id))
            .ForMember(dest => dest.title, opt => opt.MapFrom(src => src.title))
            .ForMember(dest => dest.userId, opt => opt.MapFrom(src => src.userId))
            .ForMember(dest => dest.content, opt => opt.MapFrom(src => src.content))
            .ForMember(dest => dest.created, opt => opt.MapFrom(src => src.created))
            .ForMember(dest => dest.modified, opt => opt.MapFrom(src => src.modified));
        CreateMap<NewsRequestTo, News>()
            .ForMember(dest => dest.id, opt => opt.MapFrom(src => src.id))
            .ForMember(dest => dest.title, opt => opt.MapFrom(src => src.title))
            .ForMember(dest => dest.userId, opt => opt.MapFrom(src => src.userId))
            .ForMember(dest => dest.content, opt => opt.MapFrom(src => src.content))
            .ForMember(dest => dest.created, opt => opt.MapFrom(src => src.created))
            .ForMember(dest => dest.modified, opt => opt.MapFrom(src => src.modified));
    }
    
}

File: 151003/Rustamov/publisher/src/main/java/org/example/publisher/impl/news/mapper/NewsMapper.java
package org.example.publisher.impl.news.mapper;

import org.example.publisher.impl.author.Author;
import org.example.publisher.impl.news.News;
import org.example.publisher.impl.news.dto.NewsRequestTo;
import org.example.publisher.impl.news.dto.NewsResponseTo;

import java.util.List;

public interface NewsMapper {

    NewsRequestTo newsToRequestTo(News news);

    List<NewsRequestTo> newsToRequestTo(Iterable<News> news);

    News dtoToEntity(NewsRequestTo newsRequestTo, Author author);

    NewsResponseTo newsToResponseTo(News news);

    List<NewsResponseTo> newsToResponseTo(Iterable<News> news);
}
